oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/MergeRunner.java

```diff
@@ -254,6 +274,14 @@ public void run() {
             }
         }
         log.info("Waiting for batch sorting tasks completion");
+
+        // Parent thread signals to stop immediately
+        if (unmergedFiles.contains(MERGE_FORCE_STOP_POISON_PILL)) {
+            log.info("Merger receives force stop signal, shutting down all merge tasks");
+            executorService.shutdownNow();
```

Comment:
This cancels tasks via Thread.interrupt(), which is very problematic: is closes all open files and sockets, and this can cause issues. See also https://stackoverflow.com/questions/2020992/is-thread-interrupt-evil#:~:text=The%20main%20problem%20with%20Thread,interrupt%20the%20thread%20right%20away. - and we had issues in Oak and AEM as well. 

In my view, we must not use shutdownNow here.